docs/src/variables.md

```diff
@@ -1,13 +1,97 @@
 Variables
 =========
 
-What is a JuMP VariableRef?
+What is a JuMP variable?
 ---------------------------
 
-DRAFT: A JuMP variable is a reference to an index in a model. It's a thin
-wrapper around MOI.VariableIndex. Variables have (1) names, and (2) attributes.
-Describe the different scopes of a variable (e.g., as Julia variables, lookup by
-string name, and lookup by symbol).
+The term _variable_ in computational optimization has many meanings. Here, we
+distinguish between the following three types of variables:
+1. _optimization_ variables, which are the mathematical ``x`` in the problem
+``\\max{c^\top x | Ax = b\}``.
+2. _Julia_ variables, which are bindings between a name and a value, for example
+`x = 1`. (See
+[here](https://docs.julialang.org/en/stable/manual/variables/) for the Julia
+docs.)
+3. _JuMP_ variables, which are instances of the `JuMP.VariableRef` struct
+defined by JuMP that contains a reference to an optimization variable in a
+model. (Extra for experts: the `VariableRef` struct is a thin wrapper around a
+`MOI.VariableIndex`, and also contains a reference to the JuMP model.)
+
+To illustrate these three types of variables, consider the following JuMP code
+(the full syntax is explained below):
+```julia
+model = Model()
+@variable(model, x[1:2])
+```
+This code does three things:
+1. it tells `model` to add two _optimization_ variables
+2. it creates two _JuMP_ variables that act as references to those
+optimization variables
+3. it binds those JuMP variables as a vector with two elements to the name `x`.
+
+Therefore, we have:
+```julia
+julia> x
+2-element Array{JuMP.VariableRef, 1}
+ x[1]
+ x[2]
+```
+
+To reduce confusion, we will attempt, where possible, to always refer to
+variables with their corresponding prefix.
+
+JuMP variables can have attributes, such as names or an initial primal start
+value. We illustrate the name attribute in the following example:
+```julia
+model = Model()
+@variable(model, x, basename="decision variable")
+```
+This code does four things:
+1. it tells `model` to add one _optimization_ variable
+2. it creates one _JuMP_ variable that acts as a reference to that
+optimization variables
+3. it binds the JuMP variable to the name `x`
+4. it tells JuMP that the _name_ attribute of this JuMP variable is "decision
```

Comment:
I thought when reading the previous section that the optimization variable was the MOI variable. in that case, `basename` is the name of the optimization variable isn't it ?